test/conformance/map/proto/map_checked.ts

```diff
@@ -36,10 +36,9 @@ export namespace maps {
         toObject() {
             const data: {
                 link?: string;
-            } = {};
-            if (this.link != null) {
-                data.link = this.link;
-            }
+            } = {
```

Comment:
I don't think we should set the key if the field is not set or has a default. 

`Object.keys` would still enumerate absent field names which I don't think what the current behavior is. Hence i don't think we should ever set the property if its field is absent.

Reply:
@thesayyn I think this is super-important.  You said that a field should not be present in the object returned from `toObject()` method _"if the field is not set or has a default"_.
Am I correct to understand that the defaults should not be reflected in the objects returned by `toObject()` methods and that this pull request should only affect the getters of the classes generated for protobuf messages and NOT change anything in how the `toObject()` works?

Reply:
Exactly.

Reply:
> Exactly.

Then I guess my complaint about `FieldDescriptorProto.toObject()` behaviour in https://github.com/thesayyn/protoc-gen-ts/pull/146#discussion_r895586548 is not really justified.  Anyway, the `toObject()` will probably still have to be corrected, as right now it does add default values for proto2 fields when the default values are provided explicitly in the proto file.

Reply:
and you can keep the getField call until we have absence fields. @tomasz-szypenbejl-td